First user interface, basic start

VizardAlpha opened the pull request:
First user interface.
Can be launched via the UI configuration with an argument.

See the Readme for more information.

Commit:
feat: enhance Steam module and add SteamWorkshopView

- Add fallback handling for FileService in Steam class
- Update project metadata in pom.xml (version 0.1.0)
- Implement new SteamWorkshopView UI for managing Steam Workshop mods

4rg0n commented on cli/pom.xml:
```diff
@@ -6,11 +6,11 @@
         <groupId>com.github.argon.moduploader</groupId>
         <artifactId>root</artifactId>
-        <version>local</version>
+        <version>0.1.0</version>
```
The local version will not change and always stay "local". Version will be set on build time when releasing via Github Actions. See: https://maven.apache.org/maven-release/maven-release-plugin/usage/update-versions.html

4rg0n commented on core/pom.xml:
```diff
@@ -6,11 +6,11 @@
         <groupId>com.github.argon.moduploader</groupId>
         <artifactId>root</artifactId>
-        <version>local</version>
+        <version>0.1.0</version>
```
The local version will not change and always stay "local". Version will be set on build time when releasing via Github Actions. See: https://maven.apache.org/maven-release/maven-release-plugin/usage/update-versions.html

4rg0n commented on core/src/main/java/com/github/argon/moduploader/core/vendor/steam/Steam.java:
```diff
@@ -44,7 +44,7 @@ public Steam(Integer appId, IFileService fileService, SteamMapper mapper) throws
     public Steam(Integer appId, IFileService fileService, String steamAppIdTxt, SteamMapper mapper) throws InitializeException {
         this.steamAppIdTxt = steamAppIdTxt;
-        this.fileService = fileService;
+        this.fileService = fileService != null ? fileService : new FileService();
```
This circumvents dependency injection. When the fileService is null at this point, something is wrong.

4rg0n commented on pom.xml:
```diff
@@ -3,9 +3,17 @@
     <groupId>com.github.argon.moduploader</groupId>
     <artifactId>root</artifactId>
-    <version>local</version>
+    <version>0.1.0</version>
```
The local version will not change and always stay "local". Version will be set on build time when releasing via Github Actions. See: https://maven.apache.org/maven-release/maven-release-plugin/usage/update-versions.html

VizardAlpha replied:
it was for build the .exe

4rg0n commented on pom.xml:
```diff
@@ -3,9 +3,17 @@
+    <version>0.1.0</version>
     <packaging>pom</packaging>
 
+    <name>Mod Uploader</name>
```
Don't need that yet.

4rg0n commented on ui/pom.xml:
```diff
@@ -6,9 +6,178 @@
         <groupId>com.github.argon.moduploader</groupId>
         <artifactId>root</artifactId>
-        <version>local</version>
+        <version>0.1.0</version>
```
The local version will not change and always stay "local". Version will be set on build time when releasing via Github Actions. See: https://maven.apache.org/maven-release/maven-release-plugin/usage/update-versions.html

4rg0n commented on ui/pom.xml:
```diff
@@ -6,9 +6,178 @@
+        </dependency>
+
+        <!-- For configuration persistence -->
+        <dependency>
```
This dependency is already present in the core.

4rg0n commented on ui/pom.xml:
```diff
@@ -6,9 +6,178 @@
+        </dependency>
+
+        <!-- Steam Workshop integration -->
+        <dependency>
```
This dependency is already present in the core.

4rg0n commented on ui/pom.xml:
```diff
@@ -6,9 +6,178 @@
+            </plugin>
+
+            <!-- Launch4j Maven Plugin for creating Windows executable -->
+            <plugin>
```
We use Quarkus and GraalVM for building native executables.

4rg0n commented on ui/src/main/java/com/github/argon/moduploader/ui/Launcher.java:
```diff
@@ -0,0 +1,40 @@
+            ModUploaderApplication.main(args);
+        } catch (Exception e) {
+            log.error("Failed to launch application", e);
+            System.err.println("Failed to launch application: " + e.getMessage());
```
Do not log and print to system out at the same time with the same message. Only log in this case.

4rg0n commented on ui/src/main/java/com/github/argon/moduploader/ui/Launcher.java:
```diff
@@ -0,0 +1,40 @@
+            System.err.println("Failed to launch application: " + e.getMessage());
+            
+            // If JavaFX runtime components are missing, provide helpful error message
+            if (e.getMessage() != null && e.getMessage().contains("JavaFX runtime components are missing")) {
```
Do not parse error messages and react in code. These are unreliable.

4rg0n commented on ui/src/main/java/com/github/argon/moduploader/ui/Launcher.java:
```diff
@@ -0,0 +1,40 @@
+            
+            // If JavaFX runtime components are missing, provide helpful error message
+            if (e.getMessage() != null && e.getMessage().contains("JavaFX runtime components are missing")) {
+                System.err.println("\nJavaFX runtime components are missing. Please make sure JavaFX is properly installed or use the packaged executable.");
```
Use logger

4rg0n commented on ui/src/main/java/com/github/argon/moduploader/ui/ModUploaderApplication.java:
```diff
@@ -0,0 +1,94 @@
+            launch(args);
+        } catch (Exception e) {
+            // If JavaFX runtime components are missing, try to launch with explicit module path
+            if (e.getMessage() != null && e.getMessage().contains("JavaFX runtime components are missing")) {
```
Do not parse error messages and react in code. These are unreliable.

4rg0n commented on ui/src/main/java/com/github/argon/moduploader/ui/ModUploaderApplication.java:
```diff
@@ -0,0 +1,94 @@
+                System.err.println("If running from IDE, add VM options: --module-path=\"path/to/javafx-sdk/lib\" --add-modules=javafx.controls,javafx.fxml");
+            } else {
+                log.error("Failed to launch application", e);
+                e.printStackTrace();
```
Logger already prints stacktrace

4rg0n commented on ui/src/main/java/com/github/argon/moduploader/ui/Launcher.java:
```diff
@@ -0,0 +1,40 @@
+                System.err.println("   --module-path=\"C:\\path\\to\\javafx-sdk\\lib\" --add-modules=javafx.controls,javafx.fxml,javafx.graphics");
+            }
+            
+            e.printStackTrace();
```
Logger already prints stacktrace

4rg0n commented on ui/src/main/java/com/github/argon/moduploader/ui/config/ConfigService.java:
```diff
@@ -0,0 +1,105 @@
+ * Handles loading and saving configuration to/from a JSON file.
+ */
+@Slf4j
+public class ConfigService {
```
Use Quarkus Config Mappings: https://quarkus.io/guides/config-mappings

4rg0n commented on ui/src/main/java/com/github/argon/moduploader/ui/config/UserConfig.java:
```diff
@@ -0,0 +1,40 @@
+ * This is serialized to/from JSON for persistence.
+ */
+@Data
+public class UserConfig {
```
Use Quarkus Config Mappings: https://quarkus.io/guides/config-mappings

4rg0n commented on ui/src/main/java/com/github/argon/moduploader/ui/view/SettingsView.java:
```diff
@@ -0,0 +1,323 @@
+        logLevelLabel.getStyleClass().add("subheader");
+
+        logLevelComboBox = new ComboBox<>();
+        logLevelComboBox.getItems().addAll(
```
Use Slf4J level constants

4rg0n commented on ui/src/main/java/com/github/argon/moduploader/ui/view/SettingsView.java:
```diff
@@ -0,0 +1,323 @@
+            }
+
+            // Check user home directory
+            Path userHomeLog = Paths.get(System.getProperty("user.home"), ".moduploader", "logs", "app.log");
```
Log file paths should be configurable and default next to the executable.

4rg0n reviewed (commented):
All in all it's much better than I expected xD
The biggest obstacle will be getting Quarkus to work with JavaFX:

https://quarkus.io/extensions/io.quarkiverse.fx/quarkus-fx/

VizardAlpha commented:
Ready to merge. I'll work on the gui afterwards. Check the Readme in module "ui"

Commit:
refactor: remove unused modules and configuration files

- Remove `Blockable` interface from core module
- Delete `ConfigService` class and related paths in ui module
- Remove dependency-reduced-pom.xml from ui module
- Cleanup redundant classes, resources, and FXML files for improved maintainability.
- Simplify project by removing legacy support for mod configuration and multilingual features.

- Now the application launches correctly when you run the ui module and the jar is executable when you build the module.

Commit:
Merge branch 'master' into fork/Vizard

Commit:
add resources: Mockup.epgz and readme informations

4rg0n commented on ui/src/main/java/com/github/argon/moduploader/ui/UiInitializer.java:
```diff
@@ -0,0 +1,70 @@
+@ApplicationScoped
+public class UiInitializer {
+
+    private static final Logger LOG = Logger.getLogger(UiInitializer.class);
```
Use `@Slf4j` on the class instead. This comes from lombok and gives you a `log` variable to log with.

VizardAlpha replied:
Oh I didn't know, that's a good thing. But it's still the same x)

VizardAlpha replied:
log.info("test")

4rg0n commented on ui/src/main/java/com/github/argon/moduploader/ui/controller/MainController.java:
```diff
@@ -0,0 +1,63 @@
+@Dependent
+public class MainController {
+
+    private static final Logger LOG = Logger.getLogger(MainController.class);
```
`@Slf4j` =D

4rg0n commented on ui/src/main/resources/application.properties:
```diff
@@ -0,0 +1,8 @@
+
+# Configuration pour Quarkus FX
+quarkus.fx.views-root=/fxml/
+quarkus.fx.stylesheet-reload-strategy=dev
```
if you need dev properties, profiles: https://quarkus.io/guides/config-reference#profiles

VizardAlpha replied:
Ok thanks, here it's specific to FX quarkus. But it doesn't seem to make much difference (I've just tested without it).

4rg0n commented on ui/src/main/resources/application.properties:
```diff
@@ -0,0 +1,8 @@
+# Configuration pour Quarkus FX
+quarkus.fx.views-root=/fxml/
+quarkus.fx.stylesheet-reload-strategy=dev
+quarkus.fx.source-resources=src/main/resources/
```
This is already the default path

4rg0n commented on ui/src/main/resources/application.properties:
```diff
@@ -0,0 +1,8 @@
+quarkus.fx.views-root=/fxml/
+quarkus.fx.stylesheet-reload-strategy=dev
+quarkus.fx.source-resources=src/main/resources/
+quarkus.fx.target-resources=target/classes/
```
This is already the default path

4rg0n commented on ui/src/main/resources/application.properties:
```diff
@@ -0,0 +1,8 @@
+# Configuration de l'application
+quarkus.application.name=Mon Application FX
+
+# Configuration pour Quarkus FX
```
English code docu =D

4rg0n commented on ui/src/main/resources/application.properties:
```diff
@@ -0,0 +1,8 @@
+# Configuration de l'application
```
English code docu =D

Commit:
refactor: replace `org.jboss.logging` with `lombok.extern.slf4j` across UI module

- Updated logger annotations and references to use Lombok's `@Slf4j`.
- Simplified logging configuration in `UiInitializer`, `MainController`, and `ExampleService`.
- Minor improvements in application properties for better readability.

Commit:
chore: remove unused dependencies from ui/pom.xml

- Removed `quarkus-arc` and `lombok` dependencies to simplify project configuration.
